### PR TITLE
fix typo in objectnative.cpp comments

### PR DESCRIPTION
**Pull request by romansp**

### CODE REVIEW

File: src/classlibnative/bcltype/objectnative.cpp [hunk 1]
```diff
@@ -183,7 +183,7 @@ FCIMPL2(FC_BOOL_RET, ObjectNative::Equals, Object *pThisRef, Object *pCompareRef
     if (pThisMT != pCompareRef->GetMethodTable())
         FC_RETURN_BOOL(FALSE);
 
-    // Compare the contents (size - vtable - sink block index).
+    // Compare the contents (size - vtable - sync block index).
```
Comment (jasonwilliams200OK) [inline src/classlibnative/bcltype/objectnative.cpp]: Actually, this is not a type.

**Comment by richlander** [comment]
I would have thought that was a typo. No? This seems to be the authoritative spelling: https://github.com/dotnet/coreclr/blob/master/src/vm/syncblk.h

File: src/classlibnative/bcltype/objectnative.cpp (hunk 1 above)
Comment (mikedn) [inline src/classlibnative/bcltype/objectnative.cpp]: It is a typo. Sink is a valid word but it does not belong here, sync does.

Comment (jasonwilliams200OK) [inline src/classlibnative/bcltype/objectnative.cpp]: I was looking at `ComEventsSink`, `StackBuilderSink` classes and 41 other occurrences of Sink in CoreCLR's C++ code. If you are absolutely sure, then sync that is, but please confirm.

Comment (stephentoub) [inline src/classlibnative/bcltype/objectnative.cpp]: They're two different words. A "sink" is a target or endpoint for something, a place where data end up. Here it's "sync" as in "synchronization".

**Comment by jkotas** [comment]
LGTM. Thanks!

**Pull request merged**